### ISSUE TITLE
New Feature: Admin request of Character Counts

### ISSUE THREAD

**Issue by nrshapiro** [Psychoanalytic-Electronic-Publishing/PEP-Web-User-Interface#636]
The server admin/reports have been upgraded to automatically generate the character count report and allow it to be downloaded in CSV.  But right now, the only way to do this is to go to the Stage API endpoint testbed and request it using admin/reports at https://stage-api.pep-web.org/docs#/Documents/documents_image_fetch_v2_Documents_Image__imageID___get  And you must be logged in as an admin to get those to work, so you need to know how to authenticate on that platform.

But instead, the way to do this right is to modify the admin page of the client app to allow the admin user to request the report, and download it directly from there.

Basically all you need is a button there that says something equivalent to "Download Character Counts".   The client calls the server API with a valid client id and session id for an admin, and the rest is taken care of.

Effectively, though this example will not be authorized without the client and session-id of an admin (but this example shows the essential parameters):
     https://stage-api.pep-web.org/v2/Admin/Reports/Character%20Count%20Report?sortorder=ASC&download=true

**Comment by nrshapiro** [comment]
@SophieMBennett  @jordanallen-dev @ocappello @davidtuckett 

For "Resilience"

- The server will automatically supply the text for the annual counts, including the exact paragraph we use on Wordpress, in HTML.  This could easily be added to the client admin page per the above, to make it easy for you to get to.  
- However, I think you should change the widgets on the home page to display these stats instead (at least the journals one).  The journal picture is colorful, but too small to be really informative in any way.  Putting the text there provides real marketing info.  If you get rid of the books one, that one could be taller and have an integrated picture to make it more decorative.  Since the server already provides this, it should be easy for Jordan to change.  Moreover, it will always be up to date as data is added.
- Please also note that these two infocards (widgets, as Gavant called them in the interface) are not foldable or configurable like the others, and IMHO they should be.
    - <img width="204" alt="image" src="https://user-images.githubusercontent.com/4837554/212082578-49d82cb3-7110-401f-9f3f-2e5a1b261340.png">
 - You don't need the admin report as this issue alludes to above to get this info--it can be easily accessed through the non-admin based "status" endpoint, without logging in.

Below this paragraph (below the line) is what I just downloaded from there on Production.  (It comes back as HTML, I just pasted it here and it's formatted).  You may only want the first part of this, but the other part might be of interest elsewhere, as well.)  You can see it in raw html in the interactive server API documentation or from bash or git-bash with this command: 
- curl -X 'GET' 'https://api.pep-web.org/v2/Session/Status/?moreinfo=true' -H 'accept: application/json' -H 'client-id: 4'
- - - - - - - - - -
<!DOCTYPE html><html><body><p>This release of PEP-Web contains the complete text and illustrations of 84 premier journals in psychoanalysis, 100 classic psychoanalytic books, 124 videos, and the full text and editorial notes of the 24 volumes of the Standard Edition of the Complete Psychological Works of Sigmund Freud as well as the 19 volume German Freud Standard Edition Gesammelte Werke.  It spans over 110 publication years and contains the full text of articles whose source ranges from 1912 through 2022.</p><p>There are over 144630 articles and 19667 figures and illustrations that originally resided on 2673 volumes with over 1.30 million printed pages. In hard copy, the PEP Archive represents a stack of paper more than 426 feet high and weighing over 5 tons!</p><p>The journals officially in this release (with article counts) include:<ol><li>Almanach der Psychoanalyse  - 291</li><li>American Imago - 2160</li><li>American Journal of Psychoanalysis - 3357</li><li>Annual of Psychoanalysis - 631</li><li>Articles of Interest on the Web (text searchable but only citation data returned) - 23</li><li>Attachment: New Directions in Relational Psychoanalysis and Psychotherapy - 664</li><li>British Journal of Psychotherapy - 2968</li><li>Bulletin of the American Psychoanalytical Association - 403</li><li>Bulletin of the Anna Freud Centre - 437</li><li>Bulletin of the International Psycho-Analytical Association - 700</li><li>Canadian Journal of Psychoanalysis - 1014</li><li>Chinese Annual of Psychoanalysis - 9</li><li>Contemporary Psychoanalysis - 2575</li><li>Couple and Family Psychoanalysis - 387</li><li>Division Review - 431</li><li>Fort Da - 742</li><li>Free Associations - 813</li><li>Gender and Psychoanalysis - 158</li><li>IJP Open - Open Peer Review and Debate - 678</li><li>Imago - 1210</li><li>International Forum of Psychoanalysis - 1224</li><li>International Journal of Applied Psychoanalytic Studies - 745</li><li>International Journal of Psychoanalysis  - 12040</li><li>International Journal of Psychoanalysis en Espanol - 188</li><li>International Journal of Psychoanalytic Self Psychology - 517</li><li>International Review of Psychoanalysis - 1112</li><li>Internationale Psychoanalyse - 237</li><li>Internationale Zeitschrift für Psychoanalyse - 2771</li><li>Jahrbuch der Psychoanalyse - 890</li><li>Jahrbuch für psychoanalytische und psychopathologische Forschung - 121</li><li>Journal of Analytical Psychology - 4282</li><li>Journal of Applied Psychoanalytic Studies - 223</li><li>Journal of Child Psychotherapy - 1658</li><li>Journal of Clinical Psychoanalysis - 591</li><li>Journal of Infant, Child & Adolescent Psychotherapy - 714</li><li>Journal of Psychological Therapies - 104</li><li>Journal of the American Academy of Psychoanalysis - 2325</li><li>Journal of the American Psychoanalytic Association - 5696</li><li>L'Annata Psicoanaltica Internazionale - 139</li><li>L'anneé Psychanalytique Internationale - 211</li><li>Luzifer-Amor: Zeitschrift zur Geschichte der Psychoanalyse - 1242</li><li>Modern Psychoanalysis - 1116</li><li>Neuropsychoanalysis - 921</li><li>Oedipus Annual - Greek Annual Psychoanalytic Review - 20</li><li>Organizational and Social Dynamics - 474</li><li>Progress in Self-Psychology - 400</li><li>Psicoterapia e Scienze Umane - 3140</li><li>Psikanaliz Yazıları - 555</li><li>Psyche - Zeitschrift für Psychoanalyse - 8522</li><li>Psychoanalysis Culture and Society - 1119</li><li>Psychoanalysis and Contemporary Science - 87</li><li>Psychoanalysis and Contemporary Thought - 482</li><li>Psychoanalysis and History - 534</li><li>Psychoanalysis and Psychotherapy in China - 145</li><li>Psychoanalysis, Self, and Context - 314</li><li>Psychoanalytic Books - 1203</li><li>Psychoanalytic Dialogues - 2176</li><li>Psychoanalytic Inquiry - 2283</li><li>Psychoanalytic Perspectives - 833</li><li>Psychoanalytic Psychology - 1933</li><li>Psychoanalytic Psychotherapy - 1297</li><li>Psychoanalytic Quarterly - 17103</li><li>Psychoanalytic Review - 6872</li><li>Psychoanalytic Social Work - 517</li><li>Psychoanalytic Study of the Child - 1650</li><li>Psychoanalytische Bewegung - 423</li><li>Psychodynamic Psychiatry - 587</li><li>Revista de Psicoanálisis - 1481</li><li>Revista de Psicoanálisis  - 3490</li><li>Revue Belge de Psychanalyse - 925</li><li>Revue Roumaine de Psychanalyse - 421</li><li>Revue française de psychanalyse - 9074</li><li>Revue psychoanalytická psychoterapie - 573</li><li>Rivista Italiana di Psicoanalisi - 91</li><li>Rivista di Psicoanalisi - 4156</li><li>Scandinavian Psychoanalytic Review - 967</li><li>Studies in Gender and Sexuality - 762</li><li>The International Journal of Forensic Psychotherapy - 77</li><li>The Italian Psychoanalytic Annual - 222</li><li>Tijdschrift voor psychoanalyse - 2000</li><li>Uluslararası Psikanaliz Yıllığı / The International Journal of Psychoanalysis Turkish Annual - 178</li><li>Zeitschrift für psychoanalytische Pädagogik - 753</li><li>Zentralblatt für Psychoanalyse - 1081</li><li>Ετήσια ελληνική έκδοση - 68</li></ol></p> <p>An exhaustive list of sources, including those being loaded (abbreviations listed here with article counts) include:<ol><li>Almanach d. PsA. - 291</li><li>Am. Imago - 2160</li><li>Am. J. Psychoanal. - 3357</li><li>Anna Freud Ctr. - 2</li><li>Annu. Psychoanal. - 631</li><li>Articles of Interest on the Web - 23</li><li>Attach: New Dir. Rel. Psychoanal. Psychother. - 664</li><li>Bos. Psychoanal. Soc. Inst. - 1</li><li>Brit. J. Psychother. - 2968</li><li>Bul. Amer. Psychoanal. Assn. - 403</li><li>Bul. Anna Freud Centre - 437</li><li>Bul. Int. Psychoanal. Assn. - 700</li><li>Can. J. Psychoanal. - 1014</li><li>Chi Annual of Int. J. Psychoanal. - 9</li><li>Contemp. Psychoanal. - 2575</li><li>Cpl. Fam. Psychoanal. - 387</li><li>Div. Rev. - 431</li><li>Fort Da - 742</li><li>Free Associations - 813</li><li>Gender and Psychoanal. - 158</li><li>Gesammelte Werke - 451</li><li>IJP Open - 678</li><li>Imago - 1210</li><li>Inst. Psychoanal. UK AV Proj. - 19</li><li>Int. Forum Psychoanal. - 1224</li><li>Int. J.\r\nPsychoanal. Turkish Annu. - 178</li><li>Int. J. Appl. Psychoanal. Stud. - 745</li><li>Int. J. Forens. Psychoanal. - 77</li><li>Int. J. Psychoanal. - 12040</li><li>Int. J. Psychoanal. Es. - 188</li><li>Int. J. Psychoanal. Self Psychol. - 517</li><li>Int. Psycho-Anal. Lib. - 1021</li><li>Int. Psychoanalyse - 237</li><li>Int. Rev. Psycho-Anal. - 1112</li><li>Int. Z. Psychoanal. - 2771</li><li>Ital. Psychoanal. Annu. - 222</li><li>J. Am. Acad. Psychoanal. Dyn. Psychiatr. - 2325</li><li>J. Amer. Psychoanal. Assn. - 5696</li><li>J. Anal. Psychol. - 4282</li><li>J. Appl. Psychoanal. Stud. - 223</li><li>J. Child Psychother. - 1658</li><li>J. Clin. Psychoanal. - 591</li><li>J. Infant Child Adolesc. Psychother. - 714</li><li>J. Psych. Ther. - 104</li><li>J. Revista de Psicoanal. - 1481</li><li>Jahrb. Psychoanal. - 890</li><li>Jahrb. Psychoanalyt. Psych. Forsch. - 121</li><li>L'Annata Psicoanal. Int. - 139</li><li>L'Annee Psychanal. Int. - 211</li><li>Luzifer-Amor - 1242</li><li>Mod. Psychoanal. - 1116</li><li>NY Psychoanal. Soc. Inst. - 1</li><li>Neurpsychoanalysis - 921</li><li>New Lib. of Psycho-Anal. - 125</li><li>Oed. Ann. - 20</li><li>Organ. Soc. Dyn. - 474</li><li>PEP Vid. Coll. - 23</li><li>PEP Vid. Grants - 18</li><li>PEP/UCL Top Authors Proj. - 46</li><li>Progr. Self Psychol. - 400</li><li>Psa. Books - 1203</li><li>Psikanaliz Yazıları - 555</li><li>Psyche - 8522</li><li>Psychoanal. Bew. - 423</li><li>Psychoanal. Contemp Sci - 87</li><li>Psychoanal. Contemp Thought - 482</li><li>Psychoanal. Cult. Soc. - 1264</li><li>Psychoanal. Dial. - 2176</li><li>Psychoanal. Hist. - 534</li><li>Psychoanal. Inq. - 2283</li><li>Psychoanal. Persp. - 833</li><li>Psychoanal. Psychol. - 1933</li><li>Psychoanal. Psychother. - 1297</li><li>Psychoanal. Q. - 17103</li><li>Psychoanal. Rev. - 6872</li><li>Psychoanal. Self Cxt. - 314</li><li>Psychoanal. Soc. Work - 517</li><li>Psychoanal. Study Child - 1650</li><li>Psychodyn. Psi. - 587</li><li>Psychoter. Sci. U. - 3140</li><li>Rev. Belg. Psychanal. - 925</li><li>Rev. Psicoanál.  - 3490</li><li>Rev. fr. psychanal. - 9074</li><li>Rev. psychoanal. psychoter. - 573</li><li>Rivista Italian Psicoanal. - 91</li><li>Rivista Psicoanal. - 4156</li><li>Rom. J. Psychoanal. - 421</li><li>San Fran. Ctr Psychoanal. - 4</li><li>Scand. Psychoanal. Rev. - 967</li><li>Soc. Psi. It. - 1</li><li>Standard Edition - 352</li><li>Stud. Gend. Sex. - 762</li><li>The Philoctetes Center - 2</li><li>Tijdschr. Psychoanal. - 2000</li><li>Univ. Coll. London Vid. Coll. - 2</li><li>Z. Psychoanalyt Pädagogik - 753</li><li>Zbl. Psyca - 1081</li><li>Ετήσια ελληνική έκδοση - 68</li></ol></p></body></html>
